src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -441,10 +446,138 @@ protected function getSettingsFormHandler()
+
+        $deleteExceptionMessages = [
+            CannotDeleteSqlRequestException::CANNOT_SINGLE_DELETE => $this->trans('An error occurred while deleting the object.', 'Admin.Notifications.Error'),
+            CannotDeleteSqlRequestException::CANNOT_BULK_DELETE => $this->trans('An error occurred while deleting this selection.', 'Admin.Notifications.Error'),
+        ];
+
+        if (CannotDeleteSqlRequestException::class === $type
+            && isset($deleteExceptionMessages[$code])
+        ) {
+            return $deleteExceptionMessages[$code];
+        }
+
+        if (isset($exceptionMessages[$type])) {
+            return $exceptionMessages[$type];
+        }
+
+        return $this->getFallbackErrorMessage($type, $code);
+    }
+
+    /**
+     * Get error message when exception occurs on View action.
+     *
+     * @param SqlRequestException $e
+     *
+     * @return string
+     */
+    protected function handleViewException(SqlRequestException $e)
+    {
+        $type = get_class($e);
+
+        $exceptionMessages = [
+            SqlRequestNotFoundException::class => $this->trans('The object cannot be loaded (or found)', 'Admin.Notifications.Error'),
+        ];
+
+        if (isset($exceptionMessages[$type])) {
+            return $exceptionMessages[$type];
+        }
+
+        return $this->getFallbackErrorMessage($type, $e->getCode());
+    }
+
+    /**
+     * @param Exception $e
+     *
+     * @return string Error message
+     */
+    protected function handleExportException(Exception $e)
+    {
+        $type = get_class($e);
+
+        if ($e instanceof FileWritingException) {
+            return $this->handleApplicationExportException($e);
+        }
+
+        if ($e instanceof SqlRequestException) {
+            return $this->handleDomainExportException($e);
+        }
+
+        return $this->getFallbackErrorMessage($type, $e->getCode());
+    }
+
+    /**
+     * @param FileWritingException $e
+     *
+     * @return string Error message
+     */
+    protected function handleApplicationExportException(FileWritingException $e)
+    {
+        $code = $e->getCode();
+
+        $applicationErrors = [
+            FileWritingException::CANNOT_OPEN_FILE_FOR_WRITING => $this->trans('Cannot open export file for writing', 'Admin.Notifications.Error'),
+        ];
+
+        if (isset($applicationErrors[$code])) {
+            return $applicationErrors[$code];
+        }
+
+        return $this->getFallbackErrorMessage(get_class($e), $code);
+    }
+
+    /**
+     * @param SqlRequestException $e
+     *
+     * @return string
+     */
+    protected function handleDomainExportException(SqlRequestException $e)
+    {
+        $type = get_class($e);
+
+        $domainErrors = [
+            SqlRequestNotFoundException::class => $this->trans('The object cannot be loaded (or found)', 'Admin.Notifications.Error'),
+        ];
+
+        if (isset($domainErrors[$type])) {
+            return $domainErrors[$type];
```

Comment:
this controller has too many responsibilities! Everything that is not related to HTTP layer should be moved into related services.

Reply:
see comments above regarding that.

Reply:
I don't block this pull request, but this needs to be refactored: I disagree having to maintain 600+ lines controller /c @eternoendless  this is breaking the architecture I try to push: thin controllers, fat services.

The controllers should never contain business logic, only calls to services.

Reply:
Also "How to test section" needs to be filled, please :)

Reply:
sorry, done.

Reply:
I don't (sad to see this months after).

Controllers (and it's a Symfony best practice) have one and only one responsibility: read a request and return a response.

But the most important point - that forces us to discuss the idea of error management in a controller - is that the controller is too big!

If you have only a few actions, error management is not a thing anymore and so on doesn't needs to be discussed.